source/common/upstream/outlier_detection_impl.cc

```diff
@@ -259,15 +299,44 @@ DetectorConfig::DetectorConfig(const envoy::config::cluster::v3::OutlierDetectio
       max_ejection_time_jitter_ms_(static_cast<uint64_t>(PROTOBUF_GET_MS_OR_DEFAULT(
           config, max_ejection_time_jitter, DEFAULT_MAX_EJECTION_TIME_JITTER_MS))),
       successful_active_health_check_uneject_host_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(
-          config, successful_active_health_check_uneject_host, true)) {}
+          config, successful_active_health_check_uneject_host, true)),
+      validation_visitor_(validation_visitor) {
+
+  // Store extensions' config. It will be used to create extensions monitors
+  // when host are added to the cluster.
+  extensions_config_ = config.monitors();
+}
+
+std::unique_ptr<Extensions::Outlier::MonitorsSet> DetectorConfig::createMonitorExtensions(
```

Comment:
See, reducing unnecessary heap allocation always be better. So, may be an `absl::InlinedVector<MonitorPtr, 3>` would be better.

Reply:
Done. `MonitorsSet` uses inlined vector now. Thanks for good suggestion!